internal/service/implement/auth_service.go

```diff
@@ -147,3 +155,60 @@ func (service *AuthService) ValidateRefreshToken(ctx *gin.Context, customerId in
 	}
 	return refreshToken, nil
 }
+
+func (service *AuthService) SendOTPToMail(ctx *gin.Context) error {
+	// generate otp
+	otp := mail.GenerateOTP(6)
+
+	// store otp in redis for 1 minute
+	customerId, _ := ctx.Get("customerId")
+	customerIdInt64, _ := customerId.(int64)
+	
+	baseKey := constants.REDIS_KEY
+	key := stringutils.Concat(baseKey, customerIdInt64)
+
+	err := service.redisCLient.Set(ctx, key, otp, constants.REDIS_EXP_TIME)
+	if err != nil {
+		return err
+	}
+
+	// send otp to user email
+	customerMail, err := service.customerRepository.GetMailByIdQuery(ctx, customerIdInt64)
+	err = mail.SendEmail(customerMail, "test otp", otp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (service *AuthService) ResetPassword(ctx *gin.Context, resetPasswordRequest model.ResetPasswordRequest) error {
+	customerId, _ := ctx.Get("customerId")
+	customerIdInt64, _ := customerId.(int64)
+	
+	baseKey := constants.REDIS_KEY
+	key := stringutils.Concat(baseKey, customerIdInt64)
+	
+	val, err := service.redisCLient.Get(ctx, key)
+	if err != nil {
+		return err
+	}
+
+	service.redisCLient.Delete(ctx, key)
+
+	hashedPW, err := service.passwordEncoder.Encrypt(resetPasswordRequest.Password)
+	if err != nil {
+		return err
+	}
+
+	if(val == resetPasswordRequest.OTP){
```

Comment:
not make any sense. Your flow is: get otp from redis => delete key => encrypt pw => check if otp from redis is equal to the otp in the request.
I suggest:  get otp from redis =>  check if otp from redis is equal to the otp in the request.=> delete key =>  encrypt pw